x-pack/plugins/cloud_security_posture/server/routes/compliance_dashboard/get_benchmarks.ts

```diff
@@ -0,0 +1,148 @@
+
+export interface BenchmarkBucket extends KeyDocCount {
+  aggs_by_benchmark_version: Aggregation<BenchmarkVersionQueryResult>;
+}
+
+interface BenchmarkQueryResult extends KeyDocCount {
+  aggs_by_benchmark: Aggregation<BenchmarkBucket>;
+}
+
+export type BenchmarkWithoutTrend = Omit<BenchmarkData, 'trend'>;
+
+const MAX_BENCHMARKS = 500;
+
+export const getBenchmarksQuery = (
+  query: QueryDslQueryContainer,
+  pitId: string,
+  runtimeMappings: MappingRuntimeFields
+): SearchRequest => ({
+  size: 0,
+  runtime_mappings: { ...runtimeMappings, ...getIdentifierRuntimeMapping() },
+  query,
+  aggs: {
+    aggs_by_benchmark: {
+      terms: {
+        field: 'rule.benchmark.id',
+        order: {
+          _count: 'desc',
+        },
+        size: MAX_BENCHMARKS,
+      },
+      aggs: {
+        aggs_by_benchmark_version: {
+          terms: {
+            field: 'rule.benchmark.version',
+          },
+          aggs: {
+            aggs_by_benchmark_name: {
+              terms: {
+                field: 'rule.benchmark.name',
+              },
+            },
+            asset_count: {
+              cardinality: {
+                field: 'asset_identifier',
+              },
+            },
+            // Result evalution for passed or failed findings
+            ...findingsEvaluationAggsQuery,
+            // CIS Section Compliance Score and Failed Findings
+            ...failedFindingsAggQuery,
+          },
+        },
+      },
+    },
+  },
+  pit: {
+    id: pitId,
+  },
+});
+
+export const getBenchmarksFromAggs = (benchmarks: BenchmarkBucket[]) => {
+  return benchmarks.flatMap((benchmarkAggregation: BenchmarkBucket) => {
+    const benchmarkId = benchmarkAggregation.key;
+    const versions = benchmarkAggregation.aggs_by_benchmark_version.buckets;
+    if (!Array.isArray(versions)) throw new Error('missing aggs by benchmark version');
+
+    return versions.map((version: BenchmarkVersionQueryResult) => {
+      const benchmarkVersion = version.key;
+      const assetCount = version.asset_count.value;
+      let benchmarkName = '';
+
+      if (!Array.isArray(version.aggs_by_benchmark_name.buckets))
+        throw new Error('missing aggs by benchmark name');
+
+      if (version.aggs_by_benchmark_name && version.aggs_by_benchmark_name.buckets.length > 0) {
+        benchmarkName = version.aggs_by_benchmark_name.buckets[0].key;
+      }
+      const { passed_findings: passedFindings, failed_findings: failedFindings } = version;
+      const stats = getStatsFromFindingsEvaluationsAggs({
+        passed_findings: passedFindings,
+        failed_findings: failedFindings,
+      });
+
+      const resourcesTypesAggs = version.aggs_by_resource_type.buckets;
+      if (!Array.isArray(resourcesTypesAggs))
+        throw new Error('missing aggs by resource type per benchmark');
+      const groupedFindingsEvaluation = getFailedFindingsFromAggs(resourcesTypesAggs);
```

Comment:
`getFailedFindingsFromAggs` is confusing name for a function that returns
```
{
      name: bucket.key,
      totalFindings: bucket.doc_count,
      totalFailed,
      totalPassed,
      postureScore: calculatePostureScore(totalPassed, totalFailed),
    }
```

I know it's not part of the PR, but is a good chance to rename it for more meaningful name.